ui/src/reusable_ui/components/wizard/WizardStep.tsx

```diff
@@ -0,0 +1,66 @@
+import React, {PureComponent, ReactNode} from 'react'
+import FancyScrollbar from 'src/shared/components/FancyScrollbar'
+import WizardButtonBar from 'src/reusable_ui/components/wizard/WizardButtonBar'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+import 'src/reusable_ui/components/wizard/WizardStep.scss'
+
+interface Props {
+  children: ReactNode
+  title: string
+  isComplete: () => boolean
+  onPrevious?: () => void
+  onNext?: () => void
+  increment?: () => void
+  decrement?: () => void
+  tipText?: string
+  nextLabel?: string
+  previousLabel?: string
+  lastStep?: boolean
+}
+
+@ErrorHandling
+class WizardStep extends PureComponent<Props> {
+  public render() {
+    const {children, decrement, nextLabel, previousLabel, lastStep} = this.props
+
+    return (
+      <>
+        <FancyScrollbar
+          className="step-content"
+          maxHeight={400}
+          autoHeight={true}
+          autoHide={false}
+        >
+          <div className="step-child">{children}</div>
+        </FancyScrollbar>
+        <WizardButtonBar
+          decrement={decrement}
+          nextLabel={nextLabel}
+          previousLabel={previousLabel}
+          onClickPrevious={this.handleClickPrevious}
+          onClickNext={this.handleClickNext}
+          lastStep={lastStep}
+        />
+      </>
+    )
+  }
+
+  private handleClickPrevious = async () => {
+    const {onPrevious, decrement} = this.props
+    if (onPrevious) {
+      await onPrevious()
+    }
+    decrement()
+  }
+
+  private handleClickNext = async () => {
+    const {onNext, increment} = this.props
+    if (onNext) {
+      await onNext()
+    }
+    increment()
```

Comment:
same comment as the one above. if `increment` is not required, should we ensure it's passed in before calling it?

Reply:
I'll add a check here too.